src/components/compCity/weatherHeader.js

```diff
@@ -1,8 +1,32 @@
 import TextWeatherContent from './textWeatherContent';
+import { useSession, signIn } from 'next-auth/react';
+import { useState } from 'react';
 import { HStack, VStack, Button, Center, Text } from '@chakra-ui/react';
 import { LiaStarSolid } from 'react-icons/lia';
 
-export default function WeatherHeader({ weatherData }) {
+export default function WeatherHeader({
+  weatherData,
+  addToFavorites,
+  deleteCityFromFavorites,
+  cityIsSavedToFavorites,
+}) {
+  const [isToggled, setIsToggled] = useState(false);
```

Comment:
If cityIsSavedToFavorites, coming from props, is the default value for isToggled, you want to pass it to useState as initial value, and use isToggled instead of cityIsSavedToFavorites in this component.